analysis/analysis.py

```diff
@@ -0,0 +1,297 @@
+    )
+    output_content = j2_env.get_template("chart_template.html").render(
+        gantt_div=div,
+        summary_header='Tasks matched "' + summary["name"] + '" (in ' + summary["unit"] + ")",
+        count=summary["count"],
+        min=summary["min"],
+        max=summary["max"],
+        mean=summary["mean"],
+        median=summary["median"],
+        stdev=summary["stdev"],
+        task_table=table_rows,
+    )
+
+    with open(gantt_output_file, "w") as text_file:
+        text_file.write(output_content)
+
+
+def write_csv(csv_output_file, csv_rows):
+    with open(csv_output_file, "w", newline="") as csv_file:
+        csv_writer = csv.writer(csv_file, delimiter=",", quotechar="|", quoting=csv.QUOTE_MINIMAL)
+        csv_writer.writerow(["MainTask", "SubTask", "Duration", "Hops"])
+        for r in csv_rows:
+            csv_writer.writerow(r)
+
+
+def generate_summary(csv_rows):
+    result = {}
+    # TODO: match too greedy
+    duration_transfers = list(
+        map(
+            lambda r: r[2].total_seconds(),
+            filter(lambda r: re.match("transfer", r[0], re.IGNORECASE), csv_rows),
+        )
+    )
+    data = np.array(duration_transfers)
+    description = stats.describe(data)
+    result["name"] = "Transfer"
+    result["unit"] = "Seconds"
+    result["min"] = description.minmax[0]
+    result["max"] = description.minmax[1]
+    result["mean"] = description.mean
+    result["median"] = stats.scoreatpercentile(data, 50)
+    result["stdev"] = math.sqrt(description.variance)
+    result["count"] = description.nobs
+    return result
+
+
+def write_summary(summary_output_file, summary):
+    with open(summary_output_file, "w", newline="") as summary_file:
+        json.dump(summary, summary_file)
+
+
+def fill_rows(gantt_rows, csv_rows, table_rows, task_brackets, stripped_content):
+    for task_bracket in task_brackets:
+        task_start_item = stripped_content[task_bracket[0]]
+        task_finish_item = stripped_content[task_bracket[1]]
+        task_body = task_start_item[2]["task"].split(":", 1)
+        task_id = task_start_item[2]["id"]
+        task_name = task_body[0].replace("<", "").strip() + "(#" + task_id + ")"
+        task_desc = task_body[1].replace(">", "").strip()
+        task_body_json = json.loads(task_desc.replace("'", '"'))
+
+        # add main task to rows
+        task_full_desc = json.dumps(task_body_json, sort_keys=True, indent=4).replace("\n", "<br>")
+        gantt_rows.append(
+            {
+                "Task": task_name,
+                "Start": task_start_item[0],
+                "Finish": task_finish_item[0],
+                "Description": task_full_desc,
+            }
+        )
+        hops = ""
+        if task_name.startswith("Transfer"):
+            transfer_from = task_body_json["from"]
+            transfer_to = task_body_json["to"]
+            hops = str(abs(transfer_from - transfer_to))
```

Comment:
For most cases this will be incorrect. The node index is unrelated to the channel topology.

Due to the way routing works in Raiden (either internally of via PFS, with possible refunds and re-routings due to stopped / offline nodes) it's almost impossible to extract this information from the static scenario definition.

There are scenario tasks to get the routes a PFS delivered for a certain transfer, but if they aren't included in a scenario there's not much you can do.

Reply:
I already knew that when I coded it, I think the main problem here is that we might need more information given by the logs of the scenario-player, perhaps a mapping of the nodes or a deeper object structure of the topology. Right now I have not enough knowledge of what can happen inside of Raiden during a Transfer, so I would suggest that we discuss that in a meeting, maybe?